File: photogrammetry/terrain2image_project/python/b000110002.py
from terrain2image import Point, Terr2Im

terr = Terr2Im.read('points.txt')

work = Terr2Im()

work.imageproperties = [14430, 9420, 7.2] # row, column, pixel size
work.primepoint = [-0.144, 0, 100.5] # x_0, y_0, f(c)
work.projectionpoint = [427116.06061, 4206076.87177, 1725.71198] # Xc, Yc, Zc
work.rotate = [-0.02242, 1.1115, -0.9816] # omega, phi, kappa

image = work.tr2im(terr)

pixel = work.im2px(image)

print('Terrain Coordinate')
print(terr)
print('İmage Coordinate')
print(image)
print('Pixel Coordinate')
print(pixel)

File: photogrammetry/terrain2image_project/matlab/b000110002.m
clearvars; clc

terrain = Terr2im.read('points.txt') ;
%%
work = Terr2im() ;
%%
work.improp = [14430, 9420, 7.2] ;
work.primepoint = [-0.144, 0, 100.5] ;
work.projectionpoint = [427116.06061, 4206076.87177, 1725.71198] ;
work.rotation = [-0.02242, 1.1115, -0.9816] ;
%%
M = work.M ;
%%
image = work.imagepoint(terrain) ;
%%
pixel = work.im2pix(image) ;
%%
im = imread('resim1.tif') ;

work.plot(pixel, im)